wire-library/wire-runtime-swift/src/main/swift/propertyWrappers/StringEncoded.swift

```diff
@@ -36,33 +36,14 @@ extension StringEncoded : Decodable {
             return
         }
 
-        switch decoder.stringEncodedDecodingStrategy {
-        case .allowRawDecoding:
-            guard let decodableType = Value.self as? Decodable.Type else {
-                fallthrough
-            }
-            if let stringValue = try? container.decode(String.self) {
-                let value = try Self.create(optionalEncodedValue: stringValue)
-                self.init(wrappedValue: value)
-            } else {
-                let rawValue = try container.decode(decodableType)
-                guard let value = rawValue as? Value else {
-                    throw DecodingError.typeMismatch(
-                        Value.self,
-                        DecodingError.Context(
-                            codingPath: decoder.codingPath,
-                            debugDescription: "Could not convert \(rawValue) to \(Value.self)"
-                        )
-                    )
-                }
-                self.init(wrappedValue: value)
-            }
-
-        default:
-            let stringValue = try container.decode(String.self)
+        if let stringValue = try? container.decode(String.self) {
             let value = try Self.create(optionalEncodedValue: stringValue)
             self.init(wrappedValue: value)
+            return
         }
+
+        let value = try container.decode(Value.self)
```

Comment:
Since a) all types we care about are actually codable, and b) Raw decoding is expected, I figured I might as well just require `Value` to conform to Codable